tests/FunctionsTest.php

```diff
@@ -353,6 +353,15 @@ public function testParsesResponseWithoutReason()
         $this->assertSame('Test', (string) $response->getBody());
     }
 
+	/**
+	 * @expectedException \InvalidArgumentException
```

Comment:
Add a description here. With links to https://www.rapid7.com/db/vulnerabilities/apache-httpd-cve-2016-8743 and https://tools.ietf.org/html/rfc7230#section-3.2.4

One cloud also use this paragraph: 

> Historically, HTTP header field values could be extended over multiple lines by preceding each extra line with at least one space or horizontal tab (obs-fold). A sender MUST NOT generate a message that includes line folding.

Reply:
Note that this is only deprecated in HTTP/1.1. HTTP/1.0 allows line folding. If you want to support line folding, you can just re-process the headers and replace all line folding matches with a single space.

Reply:
Ping @sagikazarmark

Reply:
I've implemented it for HTTP/1.0 in our server now: https://github.com/amphp/http-server/commit/70be9bfb53099a95328ec67a1e05983fa3c56790